src/TheThingsNetwork.cpp

```diff
@@ -199,8 +195,13 @@ int TheThingsNetwork::sendBytes(const byte* payload, int length, int port, bool
     String data = response.substring(portEnds + 1);
     int downlinkLength = data.length() / 2;
     byte downlink[64];
-    for (int i = 0, d = 0; i < downlinkLength; i++, d += 2)
-      downlink[i] = TTN_HEX_PAIR_TO_BYTE(data[d], data[d+1]);
+    byte nibble;
+    byte nextNibble;
+    for (int i = 0, d = 0; i < downlinkLength; i++, d += 2) {
+      nibble = (data[d] >= 'A') ? (data[d] - 'A' + 0x0A) : (data[d] - '0');
+      nextNibble = (data[d + 1] >= 'A') ? (data[d + 1] - 'A' + 0x0A) : (data[d + 1] - '0');
+      downlink[i] = (nibble << 4) + nextNibble;
+    }
```

Comment:
We're going to optimize this library later, so let's not replacing optimized stuff by not optimized stuff (cc @FokkeZB)

We're going back to the define `TTN_HEX_PAIR_TO_BYTE`. It won't get much faster than that. This loop is worse as it allocates to variables. You can also inline `nibble` and `nextNibble`.